### PR TITLE
Expire axes_grid1/axisartist deprecations.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by timhoffm** [comment]
There's still an error in the docs build

~~~
Traceback (most recent call last):
  File "/home/circleci/project/examples/axisartist/demo_parasite_axes.py", line 20, in <module>
    from mpl_toolkits.axisartist.parasite_axes import HostAxes, ParasiteAxes
  File "/home/circleci/project/lib/mpl_toolkits/axisartist/parasite_axes.py", line 2, in <module>
    from mpl_toolkits.axes_grid1.parasite_axes import (
ImportError: cannot import name 'parasite_axes_auxtrans_class_factory' from 'mpl_toolkits.axes_grid1.parasite_axes' (/home/circleci/project/lib/mpl_toolkits/axes_grid1/parasite_axes.py)
~~~

**Comment by anntzer** [comment]
Thanks, fixed.

**Review by timhoffm — APPROVED** [review]

File: doc/api/next_api_changes/deprecations/22098-AL.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+The following deprecated APIs have been removed:
+
+- ``SubplotDivider.figbox``, ``SubplotDivider.update_params``,
+  ``SubplotDivider.get_geometry`` (use ``get_subplotspec`` instead),
+  ``change_geometry`` (use ``set_subplotspec`` instead),
+- ``ParasiteAxesBase.update_viewlim`` (use ``apply_aspect`` instead),
+  ``ParasiteAxesAuxTransBase`` (use ``ParasiteAxesBase`` instead),
+  ``parasite_axes_auxtrans_class_factory`` (use ``parasite_axes_class_factory``
+  instead), ``ParasiteAxesAuxTrans`` (use ``ParasiteAxes`` instead),
```
Comment (QuLogic) [inline doc/api/next_api_changes/deprecations/22098-AL.rst:12]: Split apart everything with a different alternative.
```suggestion
- ``SubplotDivider.figbox``, ``SubplotDivider.update_params``,
  ``SubplotDivider.get_geometry`` (use ``get_subplotspec`` instead),
- ``change_geometry`` (use ``set_subplotspec`` instead),
- ``ParasiteAxesBase.update_viewlim`` (use ``apply_aspect`` instead),
- ``ParasiteAxesAuxTransBase`` (use ``ParasiteAxesBase`` instead),
- ``parasite_axes_auxtrans_class_factory`` (use ``parasite_axes_class_factory``
  instead),
-  ``ParasiteAxesAuxTrans`` (use ``ParasiteAxes`` instead),
```

Comment (anntzer) [reply doc/api/next_api_changes/deprecations/22098-AL.rst]: sure

**Review by QuLogic — APPROVED** [review]

**Pull request merged**